lib/blueprinter/base.rb

```diff
@@ -129,9 +129,9 @@ def self.association(method, options = {}, &block)
       name = options.delete(:name) || method
 
       options = if block_given?
-        options.merge(extractor: BlockExtractor, block: block)
+        options.merge(extractor: BlockExtractor.new, block: block)
       else
-        options.merge(extractor: AutoExtractor)
+        options.merge(extractor: AutoExtractor.new)
```

Comment:
Since you're already instantiating these `Extractor` classes, do you mind also instantiating the `Extractor` in `fields` and `identifier` 
here https://github.com/procore/blueprinter/blob/master/lib/blueprinter/base.rb#L41
and here https://github.com/procore/blueprinter/blob/master/lib/blueprinter/base.rb#L219
and then remove this https://github.com/procore/blueprinter/blob/master/lib/blueprinter/extractor.rb#L11-L13

Reply:
Removing https://github.com/procore/blueprinter/blob/master/lib/blueprinter/extractor.rb#L11-L13
makes this break: https://github.com/procore/blueprinter/blob/master/spec/integrations/shared/base_render_examples.rb#L41

This may be a breaking change, so I'm not sure if we want to change this here. Maybe I should just apply the other changes you mentioned but leave `self.extract`?

Reply:
I didn't realize there were tests for that. OK let's leave that for now. Thanks!